components/brave_shields/browser/cookie_pref_service.h

```diff
@@ -21,24 +22,37 @@ namespace brave_shields {
 class CookiePrefService : public KeyedService,
                           public content_settings::Observer {
  public:
-  explicit CookiePrefService(
-      HostContentSettingsMap* host_content_settings_map,
-      PrefService* prefs);
+  explicit CookiePrefService(HostContentSettingsMap* host_content_settings_map,
+                             PrefService* prefs);
   ~CookiePrefService() override;
 
  private:
+  class Lock {
+   public:
+    Lock();
+    ~Lock();
+    bool Try();
+    void Release();
+
+   private:
+    bool locked_;
+    DISALLOW_COPY_AND_ASSIGN(Lock);
+  };
+
   void OnPreferenceChanged();
 
   // content_settings::Observer overrides:
-  void OnContentSettingChanged(
-    const ContentSettingsPattern& primary_pattern,
-    const ContentSettingsPattern& secondary_pattern,
-    ContentSettingsType content_type,
-    const std::string& resource_identifier) override;
+  void OnContentSettingChanged(const ContentSettingsPattern& primary_pattern,
+                               const ContentSettingsPattern& secondary_pattern,
+                               ContentSettingsType content_type,
+                               const std::string& resource_identifier) override;
 
+  Lock lock_;
```

Comment:
Can we please write a detailed comment explaining this approach? It is sort of non-trivial to understand